src/system/SystemLayer.h

```diff
@@ -288,6 +290,32 @@ class LayerSockets : public Layer
     virtual SocketWatchToken InvalidSocketWatchToken() = 0;
 };
 
+class LayerSocketsLoop;
+
+/**
+ * Enables the participation of subordinate event loops in the SystemLayer event loop.
+ */
+class EventLoopHandler : public chip::IntrusiveListNodeBase<>
+{
+public:
+    virtual ~EventLoopHandler() {}
+
+    /**
+     * Prepares events and returns the next requested wake time.
```

Comment:
what events? wake up time to do what? Overall I need more documentation on this class.

Why is this in `SystemLayer` and not a separate header? Please do not increase the size of headers.

Reply:
This is not in a separate header because it belongs quite tightly to `LayerSocketsLoop`. I've clarified this in the added documentation comments.